src/saml.ts

```diff
@@ -463,54 +440,38 @@ class SAML {
   }
 
   async _requestToUrlAsync(
-    request: string | null | undefined,
+    request: string | null,
     response: string | null,
     operation: string,
     additionalParameters: querystring.ParsedUrlQuery
   ): Promise<string> {
     this.options.entryPoint = assertRequired(this.options.entryPoint, "entryPoint is required");
+    this.options.signatureAlgorithm = assertRequired(
+      this.options.signatureAlgorithm,
+      "signatureAlgorithm is required"
+    );
 
-    let buffer: Buffer;
-    if (this.options.skipRequestCompression) {
-      buffer = Buffer.from((request || response)!, "utf8");
-    } else {
-      buffer = await deflateRawAsync((request || response)!);
-    }
+    const targetUrl =
+      operation === "logout" && this.options.logoutUrl
+        ? this.options.logoutUrl
+        : this.options.entryPoint;
 
-    const base64 = buffer.toString("base64");
-    let target = new URL(this.options.entryPoint);
+    const message = request || response;
+    const messageType = request ? "SAMLRequest" : "SAMLResponse";
 
-    if (operation === "logout") {
-      if (this.options.logoutUrl) {
-        target = new URL(this.options.logoutUrl);
-      }
-    } else if (operation !== "authorize") {
-      throw new Error("Unknown operation: " + operation);
+    if (!message) {
+      throw new Error("response or request should be provided");
     }
 
-    const samlMessage: querystring.ParsedUrlQuery = request
-      ? {
-          SAMLRequest: base64,
-        }
-      : {
-          SAMLResponse: base64,
-        };
-    Object.keys(additionalParameters).forEach((k) => {
-      samlMessage[k] = additionalParameters[k];
-    });
-    if (isValidSamlSigningOptions(this.options)) {
-      if (!this.options.entryPoint) {
-        throw new Error('"entryPoint" config parameter is required for signed messages');
-      }
-
-      // sets .SigAlg and .Signature
-      this.signRequest(samlMessage);
-    }
-    Object.keys(samlMessage).forEach((k) => {
-      target.searchParams.set(k, samlMessage[k] as string);
+    return await requestToUrlAsync({
```

Comment:
Based on our previous discussion, I was expecting this to be the only call in this function. I expected to see the rest of these checks in the `requestToUrlAsync` function instead of in this `_requestToUrlAsync` function.

Reply:
Yes absolutely, that was my understanding too, I have just not did that change yet :+1:

Reply:
No worries. I'm just trying to keep up with you :)